### PR TITLE
🪟 🎉  Add frontend redirect for Cloud API token management

### LINKED ISSUE #19477: Create frontend route that redirects to speakeasy
As part of the [Public Cloud API](https://docs.google.com/document/d/1FjYyeSqCGcUW2EO0PJ5p7j5afsodM9GLSI3WBwtQHKo/edit#heading=h.5vsepilo0o5q), we need a frontend route that does the following:

1. Displays a loading state
2. Sends a request to a new Airbyte endpoint, `/speakeasy_callback_url` (authenticated request wrapped with our normal overrides)
3. Redirects the browser to the `redirectUrl` returned by the Airbyte API
4. Handles possible error states

Since the cloud API will be stood up as a new service, we need to generate a new request client using Orval based on the working OpenAPI spec:

<details>
<summary>working OpenAPI spec</summary>

```yaml
---
openapi: 3.0.2
info:
  title: Speakeasy
  version: 1.0.0
paths:
  /speakeasy_callback_url:
    summary: This Endpoint will generate a JWT that can be used to login to the Speakeasy
      developer portal. The URL that is generated from the GET contains where the
      user should be redirected to and has the JWT as in the speakeasyAccessToken
      query parameter.
    get:
      tags:
      - Speakeasy
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/RedirectUrlResponse'
          description: Returns a valid RedirectUrlResponse
        "403":
          description: If a valid JWT was not found then return a Forbidden.
      security:
      - JWT: []
      operationId: getSpeakeasyCallbackUrl
      summary: Get the Speakeasy Callback URL
      description: Return a JSON datastructure that contains the URL that should be
        redirected to in the redirectUrl field.
components:
  schemas:
    RedirectUrlResponse:
      title: Root Type for RedirectUrlResponse
      description: ""
      type: object
      properties:
        redirectUrl:
          format: url
          type: string
      example:
        redirectUrl: https://airbyte.portal.speakeasyapi.dev?speakeasyAccessToken=eydas.ad45.1234
  securitySchemes:
    JWT:
      scheme: bearer
      type: http
```

</details>

We will also need to add the service/app that provides `/speakeasy_callback_url` to the [nginx reverse proxy](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/nginx/default.conf.template)

### PR DESCRIPTION
**Pull request by joeykmh**
## What
Closes https://github.com/airbytehq/airbyte/issues/19477

Sets up a new route on cloud, `/speakeasy-redirect`, that redirects users to a [speakeasy](https://www.speakeasyapi.dev/) dashboard.

## How
1. Displays a loading state initially while the API request to `/cloud_api/speakeasy_callback_url` is resolving
2. Redirects the browser to the redirectUrl returned by the Airbyte API (with a JWT in the URL to authenticate them)
3. Handles errors by displaying an error page and a button to return to the root of the web app

### CODE REVIEW

File: speakeasy.yaml
Comment (josephkmh) [inline speakeasy.yaml:45]: Just put this yaml at the root of the project now for testing - eventually it will live somewhere in the cloud api project subdirectory, I guess.

Comment (josephkmh) [reply speakeasy.yaml]: solved by committing the generated API code instead of generating during build with orval

File: airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts:22]: This is the orval-generated API code. The Open API spec lives in `airbyte-cloud`, so I think it's more straightforward to commit directly here.

File: airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx
Comment (josephkmh) [inline airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx:29]: This has not been tested end to end yet - I have only confirmed that `<CloudApiErrorView>` is displayed. @bgroff will do some e2e testing to make sure the redirect works as expected once the backend is in place.

File: airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx
Comment (ambirdsall) [inline airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx:38]: minor nit: I'd expect a "Go back" CTA to be history-aware

File: airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx
Comment (ambirdsall) [inline airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx:47]: Too bad `AppMonitoringService.tsx` doesn't export a function type for each `AppMonitoringServiceProviderValue` field; it seems both inconvenient and brittle to have to manually-redefine this signature at the usage sites (and of course the rules of hooks prevent us from having access to `typeof trackError` in top-level module scope).

**Review by ambirdsall — COMMENTED** [review]
There's a handful of cleanup- and consistency-type changes I'd like to see, but nothing I would be uncomfortable deferring to a follow-up if merging is needed to unblock e2e testing the feature.

**Review by ambirdsall — APPROVED** [review]
Although there are some minor changes I'd like to see, I'm approving now: I'm comfortable with the user-facing changes as written (both the UI logic and the API logic LGTM; all requested changes are centered on maintainability, rather than the code's behavior), there's an internal business incentive to get it shipped by tomorrow, and it's past the end of Joey's work week.

For the record, if you can spare the time outside your business hours @josephkmh, these are the two changes I'd most want to see done before this is merged, rather than after:
- remove the noop .gitignore newline deletion
- populate the`{ apiUrl }` config object via the `useConfig` hook from `"packages/cloud/services/config"`/`"../config"` instead of hardcoding it

File: airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx
Comment (terencecho) [reply airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx:29]: we thing this is appending the redirectUrl to some base path instead of navigating directly to the external redirectUrl

File: airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts
Comment (timroes) [reply airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts:2]: ++ this file is weirdly wrong there :D (but as you mentioned outside the scope of this PR).

File: airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts
Comment (timroes) [reply airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts]: I've moved it for now into a `speakeasy` folder. Since I think we should keep the general cleanup (and merging those APIs as you suggested, which I also believe is the better actual solution) into separate cleanup.

**Comment by josephkmh** [comment]
Thanks @ambirdsall! I unfortunately don't have any bandwidth today to address the still open comments, but I'll be happy to do so once I'm back on Monday.

Thanks also @timroes for addressing the bug - looks like this is ready to merge?

File: airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts
Comment (josephkmh) [reply airbyte-webapp/src/packages/cloud/lib/domain/cloudApi/api.ts]: I think we're mixing up terms here - I intended `packages/cloud/lib/domain/cloudApi/api.ts` to be scoped to all the [Public Cloud API](https://docs.google.com/document/d/1FjYyeSqCGcUW2EO0PJ5p7j5afsodM9GLSI3WBwtQHKo/edit) endpoints. This is a new service that currently only does the speakeasy redirect, which is why that's the only endpoint there. I agree that the naming is confusing though, so I'm happy to revisit this!

File: airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts
Comment (josephkmh) [reply airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts]: This is actually not the same `cloudApiUrl` from `useConfig()` - it's a new API URL that is being introduced by the platform enablement team. Agreed that hardcoding here is not great, but it's the only place it's used currently. Happy to clean this up Monday and add a new config property once we decide on a naming scheme.

File: airbyte-webapp/src/hooks/services/AppMonitoringService/AppMonitoringService.tsx
Comment (ambirdsall) [inline airbyte-webapp/src/hooks/services/AppMonitoringService/AppMonitoringService.tsx:13]: This fixes the brittleness of manual redefining the tracking function types, which was my main practical issue before; love that. No problem merging this as-is; but FWIW, I have some thoughts about how to paint the bike shed an _even better_ color below :joy::

`AppMonitoringServiceProviderValue["trackAction"]` isn't the most ergonomic type signature to remember or type out; it makes the fact that `useAppMonitoringService` is backed by a context provider part of its public API, when I think it's really an implementation detail to be encapsulated. I think it would be a bit nicer if we split out named function types for each of the values, like
```ts
export type TrackActionFn = (actionCode: AppActionCodes, context?: Record<string, unknown>) => void;
export t pe TrackErrorFn = (error: Error, context?: Record<string, unknown>) => void;

// this could also be exported if another piece of code needs to be concerned with the full set of tracking options, but that seems like exactly the role of AppMonitoringService.tsx itself
interface AppMonitoringServiceProviderValue {
  trackAction: TrackActionFn;
  trackError: TrackErrorFn;
}
```
That way, the type defs and imports needed to addjust one of the tracking functions to a component* looks like
```ts
import { useAppMonitoringService, TrackActionFn } from "hooks/services/AppMonitoringService";

// ...

interface SpeakeasyErrorBoundaryProps {
  trackError: TrackActionFn;
}
```
which _feels_ easier to me.

\* I suspected we mainly used one or the other tracking function, rather than both, but I got curious and checked. Right now there is no application code using both in the same place:
```
┌ ~/c/airbyte/airbyte-webapp/src (master|✔) 🌓 885507f389
└➣ for f in $(rg -l trackAction); do rg -H trackError $f; done
hooks/services/AppMonitoringService/AppMonitoringService.tsx
21:  trackError: (error: Error, context?: Record<string, unknown>) => void;
46:  const trackError = (error: Error, context?: Record<string, unknown>) => {
48:      console.debug(`trackError() failed because RUM is not initialized. \n`, error);
55:  return <appMonitoringContext.Provider value={{ trackAction, trackError }}>{children}</appMonitoringContext.Provider>;
hooks/services/AppMonitoringService/__mocks__/AppMonitoringService.tsx
3:    trackError: jest.fn(),
```

File: airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts
Comment (timroes) [reply airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts]: let's work on Monday together on that. We should make sure if this is a new endpoint, that we add it properly to the env files etc so it can be env independant. We should wait for the BE to actually be deployed for this, so we can alidate that this is working properly.

File: airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts
Comment (terencecho) [inline airbyte-webapp/src/packages/cloud/services/cloudApi/useSpeakeasyRedirect.ts]: this config is incorrectly pointing the to `/cloud` prefix, instead of the intended `/cloud_api`

File: airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx
Comment (josephkmh) [reply airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx]: Yeah this is true, however just to clarify, we won't need to manually redefine the function signature at all usage sites. This is a pretty specific scenario where we can't use `useAppMonitoringService()` directly in the component since it's an error boundary and must be a class component. Everywhere else we can just do `const { trackAction, trackError } = useAppMonitoringService()` and get the types. In any case, it doesn't hurt to export the function types for this specific component!

**Pull request merged**
